src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -32,20 +33,32 @@ public class CompilerDataLogger
         public static bool TelemetryEnabled => s_telemetryClient != null;
 
         public CompilerDataLogger(IAnalysisContext analysisContext,
-                                  IEnumerable<string> targetFileSpecifiers)
+                                  IEnumerable<string> targetFileSpecifiers,
+                                  TelemetryConfiguration telemetryConfiguration = null,
+                                  TelemetryClient telemetryClient = null,
+                                  int chunkSize = 8192)
         {
-            try
+            s_telemetryConfiguration = telemetryConfiguration;
+            s_telemetryClient = telemetryClient;
+            s_sessionId = TelemetryEnabled ? Guid.NewGuid().ToString() : null;
+            ChunkSize = chunkSize;
+
+            if (!TelemetryEnabled)
```

Comment:
>if (!TelemetryEnabled) [](http://example.com/codeflow?start=12&length=22)

question, why we add this new condition

Reply:
the code in below block is to initialize a TelemetryClient instance to be used to send telemetry data.
if caller already passed in the TelemetryClient instance in constructor, we don't need to initialize a new instance.
Added the TelemetryClient / Configuration as constructor parameter so that we can overwrite the built-in instances. The unit tests can pass in mock client/configuration for testing purpose.

Reply:
also, since we are creating the compilerdatalogger for each object, we dont need to keep creating an instance of the telemetry.

with that, once it is initialized, it won't create again